File: 3349-adjacent-increasing-subarrays-detection-i/3349-adjacent-increasing-subarrays-detection-i.cpp
class Solution {
public:
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
        for (int i = 0; i + 2 * k <= n; ++i) {
            bool inc1 = true, inc2 = true;
            for (int j = i + 1; j < i + k; ++j)
                if (nums[j] <= nums[j - 1]) inc1 = false;
            for (int j = i + k + 1; j < i + 2 * k; ++j)
                if (nums[j] <= nums[j - 1]) inc2 = false;
            if (inc1 && inc2)
                return true;
        }
        return false;
    }
};

File: 3349-adjacent-increasing-subarrays-detection-i/3349-adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        for i in range(n - 2 * k + 1):
            inc1 = True
            inc2 = True
            for j in range(i + 1, i + k):
                if nums[j] <= nums[j - 1]:
                    inc1 = False
                    break
            for j in range(i + k + 1, i + 2 * k):
                if nums[j] <= nums[j - 1]:
                    inc2 = False
                    break
            if inc1 and inc2:
                return True
        return False
